test/unit/argument_parser/format_parse_validators_test.cpp

```diff
@@ -52,6 +55,24 @@ TEST(validator_test, fullfill_concept)
     EXPECT_TRUE(validator_concept<file_ext_validator>);
 }
 
+TEST(validator_test, no_file)
+{
+    filesystem::path p{"./sandbox.fasta"};
+    std::string s{"./stonebox.fasta"};
+    file_existance_validator my_validator{};
+    EXPECT_THROW(my_validator(p), parser_invalid_argument);
+    EXPECT_THROW(my_validator(s), parser_invalid_argument);
+}
```

Comment:
Can you add for both tests an actual argument parser to test the functionality in connection with the parser?
The other tests are examples for this. Something like...
```cpp
    filesystem::path file_in_path;
    std::vector<filesystem::path> path_vector;
 
     // option
     const char * argv[] = {"./argument_parser_test", "-i", "./sandbox.fasta"};
     argument_parser parser("test_parser", 3, argv);
     parser.add_option(file_in_path, 'i', "int-option", "desc",
                       option_spec::DEFAULT, file_existence_validator());
 
     EXPECT_THROW(parser.parse(), parser_invalid_argument);
```

and again with std::string. There might a compiler warning with string... If so come and get me :)